get-started/debug-and-profile.md

```diff
@@ -1,8 +1,174 @@
+
+Once the execution is paused, the context of the execution is fully preserved, and you are free to inspect the value of different variables, print the calling stack and even jump bewteen different goroutines. Once you are satisfied, you can resume the execution by stepping into the next line of code or continue the execution until the next breakpoint is encountered.
+
+Basically, the following steps are typical when you are using debugger:
+
+1. locate the code and set a breakpoint
+2. prepare data so that the execution will get through the breakpoint, and pause there as expected
+3. inspect values, follow the execution step by step
+
+### Using delve to debug test case
+
+If test cases failed, you can also use delve to debug test case. Get the name of the test case, and then use
+
+```
+dlv test -- -check.f TestName
+```
+
+at the corresponding package directory to start a debugging session that will stop at the entry of the test. For example, if you failed on `TearDownTest` in `executor/executor_test.go`, you need to get to `executor/` and run `dlv test -- -check.f TearDownTest` in your shell.
+
+### Understand how TiDB works through debugging
+
+Other than debugging a bug, it is also recommended to use the debugger for understanding how TiDB works through tracking the execution step by step.
+
+There are some functions in TiDB that are critical to understand the internals of TiDB, and you should try to pause the execution there and then run TiDB step by step to help you better understand how TiDB works.
+
+For example,
+
+1. `[executor/compiler.go:Compile](https://github.com/pingcap/tidb/blob/5c95062cc34d6d37e2e921f9bddba6205b43ee3a/executor/compiler.go#L48)` is where every SQL is compiled and optimized
+2. `[planner/planner.go:Optimize](https://github.com/pingcap/tidb/blob/5c95062cc34d6d37e2e921f9bddba6205b43ee3a/planner/optimize.go#L80)` is where the SQL optimization starts
+3. `[executor/adapter.go:ExecStmt.Exec](https://github.com/pingcap/tidb/blob/5c95062cc34d6d37e2e921f9bddba6205b43ee3a/executor/adapter.go#L312)` is where the SQL plan turns into executor, and where the SQL execution starts
+4. Each `Open`, `Next` and `Close` function of each executor marks the volcano-style execution logic
+
+As you are reading the TiDB source code, you are strongly encouraged to set a breakpoint and use the debugger to trace the execution whenever you feel confused or uncertain about the code.
+
+## Using `pprof` for profiling
+
+For any database system, performance is always an important issue, and if you want to know where the performance bottleneck is, Go provides a powerful profiling tool called `pprof` for this purpose.
+
+### Gathering runtime profiling information through HTTP end points
+
+Normally, when you have a TiDB server running, it will expose a profiling end point through HTTP at `http://127.0.0.1:10080/debug/pprof/profile`, and you can get the profile result through
```

Comment:
```suggestion
When you have a TiDB server running, normally, it will expose a profiling end point through HTTP at `http://127.0.0.1:10080/debug/pprof/profile`, and you can get the profile result by running the following commands:
```